openstack/client.go

```diff
@@ -16,6 +17,8 @@ const (
 	v30 = "v3.0"
 )
 
+var VERSION_PATTERN = regexp.MustCompile("/v[1-9]{1}[0-9.]*")
```

Comment:
I think remove the regex, because it's rarely what's needed. [At least] 2 options here:
1. extract the version, use a switch: cases 'v2.0', 'v3', 'default' . `default` returns error
2.  extract the version, strip the 'v', parse to float, check >= 2.0 and < 4.0, if not error

The first is the way to go if, as I understand it, Identity versions are static. OTOH, if users can have, say, Identity version 2.6 in the URL, then the second is the way to go.